frb_codegen/src/generator/dart/mod.rs

```diff
@@ -259,9 +299,16 @@ fn generate_dart_implementation_body(
         dart_api2wire_funcs.join("\n\n"),
         dart_api_fill_to_wire_funcs.join("\n\n"),
         dart_wire2api_funcs.join("\n\n"),
+        dart_wasm_funcs.join("\n\n"),
+        dart_wasm_module.as_deref().unwrap_or_default(),
         dart_api_impl_class_name = dart_api_impl_class_name,
         dart_wire_class_name = dart_wire_class_name,
         dart_api_class_name = dart_api_class_name,
+        external_lib_type = if wasm {
+            "WasmModule"
+        } else {
+            "ffi.DynamicLibrary"
+        }
```

Comment:
A high-level question: Currently the code is surely correct and runnable, but it may not be perfect - we seem to put logic conditioning on wasm everywhere. Is it possible to extract wasm-vs-normal target logic into some separate file/folder, and let the other files call it?

For example, in this specific example, shall we do something like 

```rust
enum Target { normal, wasm }

impl Target {
  fn external_lib_type(&self) {
    match self {
      normal => "ffiDynamicLibrary",
      wasm => "WasmModule",
    }
  }
}